modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationHolder.java

```diff
@@ -0,0 +1,246 @@
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+
+package org.nuxeo.launcher.config;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+
+import org.nuxeo.common.Environment;
+import org.nuxeo.common.codec.CryptoProperties;
+
+/**
+ * Class used to hold the configuration for Nuxeo.
+ *
+ * @since 11.5
+ */
+public class ConfigurationHolder {
+
+    /** Path representing the Nuxeo Server home directory. Usually {@code /opt/nuxeo/server}. */
+    protected final Path home;
+
+    /** Path representing the Nuxeo Server configuration file. Usually {@code /opt/nuxeo/server/bin/nuxeo.conf}. */
+    protected final Path nuxeoConf;
+
+    /**
+     * Properties representing concatenation of {@code templates/nuxeo.defaults}, {@link System#getProperties()} and
+     * {@code nuxeo.defaults} from each templates.
+     */
+    protected final Properties defaultConfig;
+
+    /** Properties representing {@code nuxeo.conf} with a fallback on {@link #defaultConfig}. */
+    protected final CryptoProperties userConfig;
+
+    protected boolean configurable;
+
+    public ConfigurationHolder(Path home) {
+        this(home, home.resolve("bin").resolve("nuxeo.conf"));
+    }
+
+    public ConfigurationHolder(Path home, Path nuxeoConf) {
+        this.home = requireNonNull(home).toAbsolutePath();
+        this.nuxeoConf = requireNonNull(nuxeoConf).toAbsolutePath();
+        // initialize some defaults
+        var basicConfig = new Properties();
+        UnaryOperator<String> homeResolver = k -> getHomePath().resolve(k).toString();
+        UnaryOperator<String> runtimeResolver = k -> getRuntimeHomePath().resolve(k).toString();
+        basicConfig.put(Environment.NUXEO_CONFIG_DIR, runtimeResolver.apply(Environment.DEFAULT_CONFIG_DIR));
+        basicConfig.put(Environment.NUXEO_DATA_DIR, runtimeResolver.apply(Environment.DEFAULT_DATA_DIR));
+        basicConfig.put(Environment.NUXEO_LOG_DIR, homeResolver.apply(Environment.DEFAULT_LOG_DIR));
+        basicConfig.put(Environment.NUXEO_MP_DIR, homeResolver.apply(Environment.DEFAULT_MP_DIR));
+        basicConfig.put(Environment.NUXEO_PID_DIR, homeResolver.apply(Environment.DEFAULT_LOG_DIR));
+        basicConfig.put(Environment.NUXEO_TMP_DIR, homeResolver.apply(Environment.DEFAULT_TMP_DIR));
+        defaultConfig = new Properties(basicConfig);
+        userConfig = new CryptoProperties(defaultConfig);
+    }
+
+    public Path getHomePath() {
+        return home;
+    }
+
+    public Path getNuxeoConfPath() {
+        return nuxeoConf;
+    }
+
+    public Path getTemplatesPath() {
+        return home.resolve("templates");
+    }
+
+    /**
+     * Returns the Home of NuxeoRuntime (same as {@code Framework.getRuntime().getHome()})
+     */
+    public Path getRuntimeHomePath() {
+        return home.resolve("nxserver");
```

Comment:
Use constant?
Also it was pointed out recently on slack doc channel that nuxeo.conf property `templateName.target` default value should not be `server/default/deploy/nuxeo.ear` but `nxserver`, but it seems this property does not exist anymore (?) 
https://doc.nuxeo.com/nxdoc/configuration-parameters-index-nuxeoconf/#templatenametarget

Reply:
Indeed the documentation should be updated to reflect that and also to reflect changes made in this PR.

The property still exists, you can find it [there](https://github.com/nuxeo/nuxeo/pull/4603/files#diff-2b3d798ff8d6cb0c3fff75b05e57d57a0f49a53e660107aa30d6a0db459a7c3cR308).